ZuluContent/Items/Resources/Fishing/Shells.cs

```diff
@@ -0,0 +1,344 @@
+using System;
+using Scripts.Zulu.Packets;
+using Server.Network;
+using Server.Spells;
+using ZuluContent.Zulu.Engines.Magic.Enchantments.Buffs;
+
+namespace Server.Items
+{
+    public abstract class BaseShrine : Item
+    {
+        public override double DefaultWeight { get; } = 1.0;
+
+        public virtual int Piece => 0;
+
+        public BaseShrine(int itemID) : base(itemID)
+        {
+        }
+
+        public BaseShrine(Serial serial) : base(serial)
+        {
+        }
+        
+        public override void Serialize(IGenericWriter writer)
+        {
+            base.Serialize(writer);
+
+            writer.Write((int) 0); // version
+        }
+
+        public override void Deserialize(IGenericReader reader)
+        {
+            base.Deserialize(reader);
+
+            int version = reader.ReadInt();
+        }
+    }
+
+    public class BaseShell : BaseShrine
+    {
+        [Constructible]
+        public BaseShell(int itemID) : base(itemID)
+        {
+        }
+
+        [Constructible]
+        public BaseShell(Serial serial) : base(serial)
+        {
+        }
+
+        public override void Serialize(IGenericWriter writer)
+        {
+            base.Serialize(writer);
+
+            writer.Write((int) 0); // version
+        }
+
+        public override void Deserialize(IGenericReader reader)
+        {
+            base.Deserialize(reader);
+
+            int version = reader.ReadInt();
+        }
+    }
+
+    public class AquariaShell : BaseShell
```

Comment:
Since these aren't created via loot we don't need individual items for each shell, what do you think about simplifying it down into one shell item and serializing `Piece`?

You would then just have a static array of the names that you could index by `Piece` as the item name, same with the ItemIds.

Reply:
Can do, done